### PR TITLE
conver servlet 5 tests from groovy to java

### PR DESCRIPTION
**Pull request by zeitlinger**
Created by copying the servlet 3 tests and then applying the diffs from between the groovy servlet 3 and groovy servlet 5 tests

### CODE REVIEW

**Comment by laurit** [comment]
We have two kinds of modules that are named `testing`. Firstly there are module that contain classes that are shared by multiple modules that contain tests, e.g. a common `testing` module for `javaagent` and `library` modules. Secondly we have modules that contain the tests because for whatever reason they can't be in the same module as the instrumentation. I don't think we ever mix the two purposes. You could also do the same here by keeping the `testing` module as shared code and moving the tomcat tests together with instrumentation or to a separate module.

File: instrumentation/servlet/servlet-5.0/testing/build.gradle.kts [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+  testInstrumentation(project(":instrumentation:jetty:jetty-11.0:javaagent"))
+
+  testImplementation(project(":instrumentation:servlet:servlet-5.0:testing"))
+
+  testLibrary("org.apache.tomcat.embed:tomcat-embed-core:10.0.0")
+  testLibrary("org.apache.tomcat.embed:tomcat-embed-jasper:10.0.0")
+
+  // Tomcat 10.1 requires Java 11
+  latestDepTestLibrary("org.apache.tomcat.embed:tomcat-embed-core:10.0.+")
+  latestDepTestLibrary("org.apache.tomcat.embed:tomcat-embed-jasper:10.0.+")
```
Comment (laurit) [inline instrumentation/servlet/servlet-5.0/testing/build.gradle.kts:21]: might as well add one more module that would test tomcat 10.1 and later

Comment (zeitlinger) [reply instrumentation/servlet/servlet-5.0/testing/build.gradle.kts:21]: we could also split the tests by java version - would reduce the number of projects from 5 to 4?

Comment (zeitlinger) [reply instrumentation/servlet/servlet-5.0/testing/build.gradle.kts:21]: or maybe we can use this - and not need to copy all test classes?

```
val testLatestDeps = findProperty("testLatestDeps") as Boolean

if (testLatestDeps) {
  otelJava {
    minJavaVersionSupported.set(JavaVersion.VERSION_11)
  }
}
```

**Comment by zeitlinger** [comment]
@laurit can you check again?

**Comment by zeitlinger** [comment]
@laurit can you check again?

**Comment by zeitlinger** [comment]
@laurit can you check again?

File: instrumentation/servlet/servlet-5.0/jetty11-testing/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  id("otel.javaagent-testing")
+}
+
+dependencies {
+  testInstrumentation(project(":instrumentation:servlet:servlet-5.0:javaagent"))
+  testInstrumentation(project(":instrumentation:jetty:jetty-8.0:javaagent"))
+  testInstrumentation(project(":instrumentation:jetty:jetty-11.0:javaagent"))
+  testImplementation(project(":instrumentation:servlet:servlet-5.0:testing"))
+
+  // Jetty 11 requires Java 11
```
Comment (laurit) [inline instrumentation/servlet/servlet-5.0/jetty11-testing/build.gradle.kts:11]: this comment feels misplaced, perhaps it should be moved to `otelJava`?

File: instrumentation/servlet/servlet-5.0/jetty11-testing/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  testLibrary("org.eclipse.jetty:jetty-server:11.0.0")
+  testLibrary("org.eclipse.jetty:jetty-servlet:11.0.0")
+
+  latestDepTestLibrary("org.eclipse.jetty:jetty-server:11.+")
```
Comment (laurit) [inline instrumentation/servlet/servlet-5.0/jetty11-testing/build.gradle.kts:15]: could you add a comment here ` // see jetty-12-testing module` like we have elsewhere

File: instrumentation/servlet/servlet-5.0/tomcat-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/tomcat/TomcatServlet5Test.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+  protected boolean hasResponseSpan(ServerEndpoint endpoint) {
+    return endpoint == NOT_FOUND || super.hasResponseSpan(endpoint);
+  }
+
+  @Override
+  protected Tomcat setupServer() throws Exception {
+    Tomcat tomcatServer = new Tomcat();
+
+    File baseDir = Files.createTempDirectory("tomcat").toFile();
+    baseDir.deleteOnExit();
```
Comment (laurit) [inline instrumentation/servlet/servlet-5.0/tomcat-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/tomcat/TomcatServlet5Test.java:95]: Not really caused by the conversion, but for this to work the directory would need to be empty.

Comment (laurit) [reply instrumentation/servlet/servlet-5.0/tomcat-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/tomcat/TomcatServlet5Test.java:95]: Hopefully fixed with https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12455

File: instrumentation/servlet/servlet-5.0/tomcat-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/tomcat/mapping/TomcatServlet5MappingTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import java.io.File;
+import java.nio.file.Files;
+import java.util.UUID;
+import org.apache.catalina.Context;
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.startup.Tomcat;
+import org.junit.jupiter.api.condition.EnabledForJreRange;
+import org.junit.jupiter.api.condition.JRE;
+
+@EnabledForJreRange(min = JRE.JAVA_11)
```
Comment (laurit) [inline instrumentation/servlet/servlet-5.0/tomcat-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/tomcat/mapping/TomcatServlet5MappingTest.java:19]: groovy test doesn't seem to have this restriction

**Review by laurit — APPROVED** [review]

**Review by jaydeluca — APPROVED** [review]

**Pull request merged**